mailbox/event/json/src/main/scala/org/apache/james/event/json/EventSerializer.scala

```diff
@@ -50,16 +48,15 @@ private object DTO {
   }
 
   case class QuotaUsageUpdatedEvent(user: User, quotaRoot: QuotaRoot, countQuota: Quota[QuotaCount],
-                                    sizeQuota: Quota[QuotaSize], time: Instant) extends QuotaEvent {
-    override def getQuotaRoot: QuotaRoot = quotaRoot
+                                    sizeQuota: Quota[QuotaSize], time: Instant) extends Event {
+    def getQuotaRoot: QuotaRoot = quotaRoot
 
-    override def toJava: JavaQuotaEvent =
+    override def toJava: JavaEvent =
       new JavaQuotaUsageUpdatedEvent(user, getQuotaRoot, countQuota.toJava, sizeQuota.toJava, time)
   }
-
 }
 
-private object JsonSerialize {
+private class JsonSerialize(mailboxIdFactory: MailboxId.Factory) {
   implicit val userWriters: Writes[User] = (user: User) => JsString(user.asString)
   implicit val quotaRootWrites: Writes[QuotaRoot] = quotaRoot => JsString(quotaRoot.getValue)
```

Comment:
You have to extract the `Quota` parts in files as @mbaechler showed in the mod programming.

Now your module is generic and should not have specific parts (except the unmanaged code).

Reply:
Well, what's generic?

I just went from static to dynamic as I need to access the MailboxId factory

The alternative is DTOing on java side in order to let scala handle **only** strings and not MailboxId. That is clearly what we try to avoid.

Reply:
> Can we, for now, accept that this file groups all Events ?

:+1: 

> Otherwise, we can have one file by ADT and accept some duplication ?

:-1: the all approache is to avoid duplication. I fear we would loose the benefits.

Reply:
what is the value of having Event a sealed trait?

Reply:
I have had a long meeting with @chibenwa ok for me

Reply:
@rouazana To have a close set of value, that help the derivation working.